apps/api/src/secret/service/secret.service.ts

```diff
@@ -502,62 +505,28 @@ export class SecretService {
         }
       }
     }
-
     // Group secrets by environment
-    const secretsByEnvironment = await secrets.reduce(
-      async (accPromise, secret) => {
-        const acc = await accPromise
-        const environmentName = await this.getEnvironmentName(
-          secret.environmentId
-        )
-
-        if (!acc[secret.environmentId]) {
-          acc[secret.environmentId] = {
-            environmentName: environmentName,
-            environmentId: secret.environmentId,
+    const secretsByEnvironment = secrets.reduce((acc, secret) => {
```

Comment:
You can get rid of this entire block and do the thing as i mentioned in the review as that will be much more clean and efficient